### PR TITLE
[SPARK-27402][INFRA][FOLLOW-UP] Exclude 'hive-thriftserver' in modules to test for hadoop3.2 for now

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR excludes  'hive-thriftserver' in modules to test for hadoop3.2 for now as well

## How was this patch tested?

Manually tested via `run-tests.py`

### CODE REVIEW

File: dev/run-tests.py [hunk 1]
```diff
@@ -112,6 +112,7 @@ def determine_modules_to_test(changed_modules):
     ['graphx', 'examples']
     >>> x = [x.name for x in determine_modules_to_test([modules.sql])]
     >>> x # doctest: +NORMALIZE_WHITESPACE
+    ...   # doctest: +SKIP
     ['sql', 'avro', 'hive', 'mllib', 'sql-kafka-0-10', 'examples', 'hive-thriftserver',
```
Comment (HyukjinKwon) [inline dev/run-tests.py]: l manually tested with `HADOOP_PROFILE=hadoop3.2 python -m doctest run-tests.py` and `python -m doctest run-tests.py`

File: dev/sparktestsupport/modules.py [hunk 2]
```diff
@@ -568,6 +558,15 @@ def __hash__(self):
...
 )
 
+# TODO: Skip hive-thriftserver module for hadoop-3.2. remove this once hadoop-3.2 support it
+if os.environ.get("AMPLAB_JENKINS"):
+    hadoop_version = os.environ.get("AMPLAB_JENKINS_BUILD_PROFILE", "hadoop2.7")
+else:
+    hadoop_version = os.environ.get("HADOOP_PROFILE", "hadoop2.7")
+if hadoop_version == "hadoop3.2":
+    print("[info] Skip unsupported module:", "hive-thriftserver")
+    all_modules = [m for m in all_modules if m.name != "hive-thriftserver"]
```
Comment (HyukjinKwon) [inline dev/sparktestsupport/modules.py]: Similarily, I tested `all_modules`. This is just to move the codes to remove later in single place.

**Comment by dongjoon-hyun** [comment]
Thank you so much, @HyukjinKwon !

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 3]
```diff
@@ -1770,7 +1770,7 @@ object SQLConf {
 }
 
 /**
- * A class that enables the setting and getting of mutable config parameters/hints.
+ * A class that enables the setting and getting of mutable config parameters/hints
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: This is for trigger the test, right?

Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Yup. I will revert..

File: dev/sparktestsupport/modules.py [hunk 4]
```diff
@@ -21,12 +21,6 @@
 import re
 import os
 
-if os.environ.get("AMPLAB_JENKINS"):
-    hadoop_version = os.environ.get("AMPLAB_JENKINS_BUILD_PROFILE", "hadoop2.7")
-else:
-    hadoop_version = os.environ.get("HADOOP_PROFILE", "hadoop2.7")
-print("[info] Choosing supported modules with Hadoop profile", hadoop_version)
-
```
Comment (dongjoon-hyun) [inline dev/sparktestsupport/modules.py]: Since this is a partial revert of SPARK-27402, ping @wangyum .

File: dev/run-tests.py [hunk 5]
```diff
@@ -122,9 +123,15 @@ def determine_modules_to_test(changed_modules):
     # If we need to run all of the tests, then we should short-circuit and return 'root'
     if modules.root in modules_to_test:
         return [modules.root]
-    return toposort_flatten(
+    changed_modules = toposort_flatten(
         {m: set(m.dependencies).intersection(modules_to_test) for m in modules_to_test}, sort=True)
 
+    # TODO: Skip hive-thriftserver module for hadoop-3.2. remove this once hadoop-3.2 support it
```
Comment (wangyum) [inline dev/run-tests.py]: Could we revert modules.py to https://github.com/apache/spark/blob/3729efb4d0420700a396c79a83a1d5db25ac3bcb/dev/sparktestsupport/modules.py and just keep this change is enough?

Comment (HyukjinKwon) [reply dev/run-tests.py]: Just keeping this change isn't enough because build and some other checks use profiles from `root`'s `build_profile_flags`.

Comment (wangyum) [reply dev/run-tests.py]: OK. Thank you @HyukjinKwon

Comment (HyukjinKwon) [reply dev/run-tests.py]: To be clear, this fix itself is a temp fix too in order to make the affected scope isolated and minimised. To fix it properly, we should maybe be able to fix it within `Module` at `modules.py`.

**Comment by HyukjinKwon** [comment]
profiles look fine so far. I will switch to SBT's to make sure it doesn't affect our regular builds.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
okie profiles look good.

File: dev/sparktestsupport/modules.py (hunk 2 above)
Comment (viirya) [reply dev/sparktestsupport/modules.py]: This just does the same thing as before, right?

Comment (HyukjinKwon) [reply dev/sparktestsupport/modules.py]: yea I just moved.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by wangyum** [comment]
@HyukjinKwon May be need add `[test-maven]` in title because of [HIVE-21639](https://issues.apache.org/jira/browse/HIVE-21639).

**Comment by HyukjinKwon** [comment]
Do you mean regular PR build with hadoop 2.7 is affected?

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by wangyum** [comment]
hadoop-3.2 needs `[test-maven]`. hadoop-2.7 does not `[test-maven]`.

**Comment by HyukjinKwon** [comment]
yea thanks for clarification.

**Comment by HyukjinKwon** [comment]
Okie. I think it's ready if I am not mistaken.

File: dev/sparktestsupport/modules.py (hunk 2 above)
Comment (dongjoon-hyun) [reply dev/sparktestsupport/modules.py]: Got it. Thank you.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.

**Comment by HyukjinKwon** [comment]
Thanks, @dongjoon-hyun

**Pull request closed without merging**